router/types/types.go

```diff
@@ -74,23 +75,24 @@ func (dj *DestinationJobT) JobIDs() map[int64]struct{} {
 
 // JobMetadataT holds the job metadata
 type JobMetadataT struct {
-	UserID             string          `json:"userId"`
-	JobID              int64           `json:"jobId"`
-	SourceID           string          `json:"sourceId"`
-	SourceCategory     string          `json:"sourceCategory"`
-	DestinationID      string          `json:"destinationId"`
-	AttemptNum         int             `json:"attemptNum"`
-	ReceivedAt         string          `json:"receivedAt"`
-	CreatedAt          string          `json:"createdAt"`
-	FirstAttemptedAt   string          `json:"firstAttemptedAt"`
-	TransformAt        string          `json:"transformAt"`
-	WorkspaceID        string          `json:"workspaceId"`
-	Secret             json.RawMessage `json:"secret"`
-	JobT               *jobsdb.JobT    `json:"jobsT,omitempty"`
-	WorkerAssignedTime time.Time       `json:"workerAssignedTime"`
-	DestInfo           json.RawMessage `json:"destInfo,omitempty"`
-	DontBatch          bool            `json:"dontBatch"`
-	TraceParent        string          `json:"traceparent"`
+	UserID             string                    `json:"userId"`
+	JobID              int64                     `json:"jobId"`
+	SourceID           string                    `json:"sourceId"`
+	SourceCategory     string                    `json:"sourceCategory"`
+	DestinationID      string                    `json:"destinationId"`
+	AttemptNum         int                       `json:"attemptNum"`
+	ReceivedAt         string                    `json:"receivedAt"`
+	CreatedAt          string                    `json:"createdAt"`
+	FirstAttemptedAt   string                    `json:"firstAttemptedAt"`
+	TransformAt        string                    `json:"transformAt"`
+	WorkspaceID        string                    `json:"workspaceId"`
+	Secret             json.RawMessage           `json:"secret"`
+	JobT               *jobsdb.JobT              `json:"jobsT,omitempty"`
+	WorkerAssignedTime time.Time                 `json:"workerAssignedTime"`
+	DestInfo           json.RawMessage           `json:"destInfo,omitempty"`
+	DontBatch          bool                      `json:"dontBatch"`
+	TraceParent        string                    `json:"traceparent"`
+	Parameters         routerutils.JobParameters `json:"parameters"`
```

Comment:
Isn't this introducing duplicate information, e.g `SourceID`, `DestinationID`, etc?
Isn't `JobMetadataT` sent/received to/from the transformer as well? Do we need this extra complexity of increasing the payload size and relying on transformer to return it back intact?

Reply:
Yeah that was my only concern of introducing duplicate information but this was needed to skip unmarshalling params at status update for every job and since we unmarshal the transformer response anyways we would get away with another unmarshal and I did not remove the existing fields just because they might be used in Transformer for some reason

I also looked into recreating Parameters from the existing fields of metadata but there were too many missing fields

Reply:
What do we really gain if we anyways pay the cost of marshaling/unmarshaling the parameters in the transformer's request? Can't we avoid that?

Reply:
If we just add more fields into the metadata struct, we can just recreate the parameters from the response but wouldn't it be the same as passing params and unmarshalling it?

Reply:
I would refrain from adding anything to this metadata struct, it is a weak design to use service calls for carrying over information which is irrelevant to the service call itself.
I prefer to keep things it they were, at least until we find a better design that doesn't entail carrying over information through HTTP calls.

Reply:
Can we use the Dehydrate/Hydrate for parameters?